portal-kernel/src/com/liferay/portal/kernel/util/SubscriptionSender.java

```diff
@@ -660,12 +657,20 @@ protected Boolean hasSubscribePermission(
 				subscription.getClassName(), ActionKeys.SUBSCRIBE);
 
 		if (resourceAction != null) {
+			ModelResourcePermission<?> modelResourcePermission =
+				ModelResourcePermissionRegistryUtil.getModelResourcePermission(
+					subscription.getClassName());
+
+			if (modelResourcePermission == null) {
+				return false;
+			}
+
 			return BaseModelPermissionCheckerUtil.containsBaseModelPermission(
-				permissionChecker, groupId, subscription.getClassName(),
+				modelResourcePermission, permissionChecker, groupId,
 				subscription.getClassPK(), ActionKeys.SUBSCRIBE);
 		}
 
-		return Boolean.TRUE;
+		return true;
 	}
 
 	protected void notifyPersistedSubscriber(
```

Comment:
@W2jx, we may simplify the logic as the following:
```
if (resourceAction == null) {
  return true;
}

ModelResourcePermission<?> modelResourcePermission =
	ModelResourcePermissionRegistryUtil.getModelResourcePermission(
			subscription.getClassName());
...

return BaseModelPermissionCheckerUtil.containsBaseModelPermission();
```

Reply:
OK